sos/report/__init__.py

```diff
@@ -903,15 +916,21 @@ def _collect_plugin(self, plugin):
         against the plugin as a whole"""
         with ThreadPoolExecutor(1) as pool:
             try:
+                _plug = self.loaded_plugins[plugin[0]-1][1]
                 t = pool.submit(self.collect_plugin, plugin)
                 # Re-type int 0 to NoneType, as otherwise result() will treat
                 # it as a literal 0-second timeout
-                timeout = self.loaded_plugins[plugin[0]-1][1].timeout or None
+                timeout = _plug.timeout or None
+                start = datetime.now()
+                _plug.manifest.add_field('start_time', start)
                 t.result(timeout=timeout)
+                end = datetime.now()
+                _plug.manifest.add_field('end_time', end)
+                _plug.manifest.add_field('run_time', end - start)
```

Comment:
A note from an English forever-noob: the `[start|end|run]_time` is well human understandable, but logically incoherent with `setup_[start|end|time]`. But I failed to find out labels that are both logically coherent and such straighforwardly understandable :-/ .

Reply:
Yeah I don't know of a better way to phrase it, but at the same time I don't think this will be a real issue seeing as the contributors would be the only ones interested in the `setup_` fields, and once you're familiar with the fact that plugins run `setup()` before collection, it tracks easily (at least in my mind).